deploy/tests/test_utils.py

```diff
@@ -1,37 +1,50 @@
 from pathlib import Path
 from unittest.mock import MagicMock, call
-from util.fsspec_utils import downloadURI
+from util.io import downloadURI
 from fsspec.implementations.local import LocalFileSystem
+from botocore.exceptions import ClientError
+import pytest
 
+pytest.mark.usefixtures("asw_test_credentials")
 
-def test_download_uri(mocker):
-    logger_mock = mocker.patch("util.fsspec_utils.rootLogger", MagicMock())
-    local_test_path = Path("tests/spec_results_download.csv")
-    test_uri = "https://www.spec.org/cpu2017/results/res2018q1/cpu2017-20171224-02028.csv"
+
+def test_download_s3_uri(mocker, mock_s3_client):
+    logger_mock = mocker.patch("util.io.rootLogger", MagicMock())
+    test_file_path = Path("tests/s3_test_json.json")
+    test_dest_file_path = Path("tests/s3_test_download_json.json")
+    test_bucket = "TestBucket"
+    test_bucket_key = "s3_blob.json"
+
+    try:
+        mock_s3_client.create_bucket(Bucket="TestBucket")
+        mock_s3_client.upload_file(str(test_file_path), test_bucket, test_bucket_key)
+        file_uri = f"s3://{test_bucket}/{test_bucket_key}"
+    except ClientError as e:
+        pytest.fail("Failed to mock an S3 client and upload a file.")
 
     fs = LocalFileSystem()
 
-    if local_test_path.exists():
-        fs.rm(str(local_test_path))
+    if test_dest_file_path.exists():
+        fs.rm(str(test_dest_file_path))
 
     downloadURI(
```

Comment:
What is causing fsspec to use your `mock_s3_client`?  In the past, when testing things that don't explicitly take a client parameter, I've had to `patch.object` the `boto3._get_default_session` as in https://github.com/firesim/firesim/blob/e2e69dbb3bb6984c3b3017bae698c0d125205ec4/deploy/tests/awstools/test_awstools.py#L62-L71 to get the code under test to  use the mock client.  Perhaps my tests are doing something unnecessary though.

Reply:
I believe that when you use moto's mock_s3() as a decorator, it causes fsspec to also instantiate mocked s3 clients, basically the same client that's instantiated in setting up in the integration test.

Reply:
https://buildmedia.readthedocs.org/media/pdf/moto/latest/moto.pdf

I was reading these docs, and they give a good example on page 6 (page 10 of the pdf)

Reply:
I believe you've had to do an extra patch because you wanted to customize how your client behaves (return an authorization error).

Reply:
Yeah, that's a good point.  I was misremembering why I needed  to do that.